islandora.install

```diff
@@ -132,3 +132,18 @@ function islandora_update_7001(&$sandbox) {
   $t = get_t();
   return $t("Islandora database updates complete");
 }
+
+/**
+ * Implements hook_update_N().
+ *
+ * Add default permissions for viewing inactive or deleted objects.
+ * Match permissions for viewing all objects.
+ */
+function islandora_update_7002(&$sandbox) {
```

Comment:
I'm uncomfortable with this update hook. I think it violates the principal of least surprise. This isn't how most Drupal modules handle permission changes. I think there are a couple cases that could happen here: 
1) The user doesn't care about this permission and so it should work like any other new permission (not get applied in a update hook). 
2) The user does care about this permission and needs to update their roles. I think in this case the user should update the roles by hand, so that the next role they create gets the correct permissions. Otherwise when creating new roles it will be surprising that this permission is there and needs to be granted. 

Based on this, I would rather see the update hook just return the message that it already does, maybe with a little more detail, and remove the bit that automatically updates permissions.